lib/afinter.c

```diff
@@ -335,6 +336,10 @@ afinter_sd_init(LogPipe *s)
   if (!log_src_driver_init_method(s))
     return FALSE;
 
+  /* If live collection of internal messages was on, and if we reload
+   * configuration having internal source, then live collection should be stopped. */
+  afinter_reset_live_collection();
```

Comment:
Calling this unconditionally will clear the internal source's queue's content even if no "live collection" is running, and the internal source is simply used elsewhere.

Reply:
@jszigetvari : Now, I have put the live_collection flag check inside the ```afinter_reset_live_collection``` function.